### ISSUE TITLE
Import GPX Tracks

### ISSUE THREAD

**Issue by HeaDBanGer84** [y20k/trackbook#36]
First want to thank you for the great app.

Would it be possible to import gpx data frim other apps? For example from the share menu.

**Comment by Hob111** [comment]
I agree, a very good app.
It would be nice if, when exporting files, one could share them too. At the moment I have to email files in two stages.

**Comment by y20k** [comment]
@HeaDBanGer84 I personally think that Trackbook is not the nicest GPX viewer. But the ability to accept GPX files and import them would be a nice feature. I am a bit reluctant, because I never wrote (or customized) an XML parser. I will look into it, but do not get your hopes up yet.

**Comment by y20k** [comment]
@Hob111 Good idea. A share sheet is probably the more native implementation of the GPX export feature than the download. I moved the request to a different issue (see #37)

**Comment by Hob111** [comment]
@y20k Thanks for creating issue #37. I agree that Trackbook isn't the nicest GPX viewer but that isn't what it's about. It does what it's designed to do very well:- record walks, bike rides etc in a standard, shareable format. There are plenty of other ways to view GPX files, edit them etc, why clutter Trackbook with unneeded features? But maybe you relish the challenge of writing an XML parser? 😃

**Comment by Hob111** [comment]
@HeaDBanGer84 Do you have a special reason for wanting to import GPX files? See my comment to @y20k. Maybe there's an advantage I've overlooked?

**Comment by HeaDBanGer84** [comment]
[Gadgetbridge](https://github.com/Freeyourgadget/Gadgetbridge) can share gps data for recorded activities and it would be nice to view it in Trackbook.

**Comment by Hob111** [comment]
I remain unconvinced. Trackbook wasn't conceived as a GPX viewer and isn't a particularly flexible one as there is only one viewing mode and no editing feature. Those who've chosen other ways to record their activities need to use a more appropriate method to view them. As I've said before, why clutter Trackbook with unneeded features? I export my walks to a laptop where there are a multitude of different ways to view and process them.
